tsdb/chunks/head_chunks_test.go

```diff
@@ -156,25 +156,30 @@ func TestChunkDiskMapper_WriteChunk_Chunk_IterateChunks(t *testing.T) {
 // TestChunkDiskMapper_Truncate tests
 // * If truncation is happening properly based on the time passed.
 // * The active file is not deleted even if the passed time makes it eligible to be deleted.
-// * Empty current file does not lead to creation of another file after truncation.
 // * Non-empty current file leads to creation of another file after truncation.
 func TestChunkDiskMapper_Truncate(t *testing.T) {
-	hrw := testChunkDiskMapper(t)
+	hrw := createChunkDiskMapper(t, "")
 	defer func() {
 		require.NoError(t, hrw.Close())
 	}()
 
 	timeRange := 0
 	fileTimeStep := 100
 	var thirdFileMinT, sixthFileMinT int64
-
+	var callbackWg sync.WaitGroup
 	addChunk := func() int {
+		t.Helper()
+
+		callbackWg.Add(1)
```

Comment:
Similar to what you did in another test with `awaitCb := make(chan struct{})`, can we do the same here and get rid of this shared `callbackWg`?

Reply:
I replaced it everywhere in 575ff12b8